commcare_connect/opportunity/visit_import.py

```diff
@@ -311,3 +313,121 @@ def get_status_by_completed_work_id(dataset):
+    return _bulk_update_catchments(opportunity, imported_data)
+
+
+def _bulk_update_catchments(opportunity: Opportunity, dataset: Dataset):
+    headers = [header.lower() for header in dataset.headers or []]
+    if not headers:
+        raise ImportException("The uploaded file did not contain any headers")
+
+    latitude_index = _get_header_index(headers, LATITUDE_COL)
+    longitude_index = _get_header_index(headers, LONGITUDE_COL)
+    active_index = _get_header_index(headers, ACTIVE_COL)
+    radius_index = _get_header_index(headers, RADIUS_COL)
+    area_name_index = _get_header_index(headers, AREA_NAME_COL)
+
+    with transaction.atomic():
+        to_create = []
+        to_update = []
+        invalid_rows = []
+        seen_catchments = set()
+        missing_catchments = 0
+
+        opportunity_access = {}
+        username_index = None
+        if USERNAME_COL in headers:
+            username_index = _get_header_index(headers, USERNAME_COL)
+            opportunity_access = {
+                oa.user.username: oa
+                for oa in OpportunityAccess.objects.filter(opportunity=opportunity).select_related("user")
+            }
+
+        for row in dataset:
+            row = list(row)
+            try:
+                latitude = Decimal(row[latitude_index])
+                longitude = Decimal(row[longitude_index])
+                radius = int(row[radius_index])
+                active = row[active_index].lower().strip() == "yes"
+                area_name = str(row[area_name_index])
+
+                if latitude < Decimal("-90") or latitude > Decimal("90"):
+                    raise ValueError("Latitude must be between -90 and 90 degrees")
+
+                if longitude < Decimal("-180") or longitude > Decimal("180"):
+                    raise ValueError("Longitude must be between -180 and 180 degrees")
+
+                if not radius:
+                    raise ValueError("Radius must be an integer")
+
+                if active_index is None or row[active_index].lower().strip() not in ["yes", "no"]:
+                    raise ValueError("Active status must be 'yes' or 'no'")
+
+                if area_name_index is None or not isinstance(row[area_name_index], str):
+                    raise ValueError("Area name is not valid.")
+
+                if not username_index or not row[username_index]:
+                    catchment = CatchmentArea(
+                        latitude=latitude,
+                        longitude=longitude,
+                        radius=radius,
+                        opportunity=opportunity,
+                        name=area_name,
+                        active=active,
+                    )
+                    missing_catchments += 1
+                    to_create.append(catchment)
+                elif row[username_index] in opportunity_access:
+                    username = row[username_index]
+                    catchment = None
+                    created = None
```

Comment:
nit: since this is a boolean elsewhere, it can just be false

Reply:
Fixed